lnd.go

```diff
@@ -755,8 +753,28 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 
 // getTLSConfig returns a TLS configuration for the gRPC server and credentials
 // and a proxy destination for the REST reverse proxy.
-func getTLSConfig(cfg *Config) (*tls.Config, *credentials.TransportCredentials,
-	string, func(), error) {
+func getTLSConfig(cfg *Config) ([]grpc.ServerOption, []grpc.DialOption,
+	func(net.Addr) (net.Listener, error), func(), error) {
+
+	// For our REST dial options, we'll increase the max message size that
+	// we'll decode to allow clients to hit endpoints which return more
+	// data such as the DescribeGraph call.  We set this to 200MiB atm.
+	// Should be the same value as maxMsgRecvSize in cmd/lncli/main.go.
+	var serverOpts []grpc.ServerOption
+	restDialOpts := []grpc.DialOption{
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(1 * 1024 * 1024 * 200),
+		),
+	}
+
+	// If TLS is disabled, we can immediately return, as no credentials
+	// have to be added.
+	if cfg.DisableTLS {
```

Comment:
Don't we need a `grpc.WithInsecure()` here?
Also disabling TLS on gRPC has an impact on `lncli`. We need a `--notls` flag there as well.
And we also need to think about the interactions between this flag and the `--no-macaroons` flag. For example the gRPC client we use for `lncli` will refuse to send the macaroon over a non-TLS connection.

Reply:
Made the change REST only.